controllers/testdata/deploy/vllm-multinode-servingruntime.yaml

```diff
@@ -0,0 +1,320 @@
+              else
+                if ray health-check --address "${HEAD_SVC}.${POD_NAMESPACE}.svc.cluster.local:6379"; then
+                  echo "Global Control Service(GCS) is ready. Any error messages above can be safely ignored."
+                  break
+                fi
+                echo "$SECONDS seconds elapsed: Still waiting for Global Control Service(GCS) to be ready."
+                echo "For troubleshooting, refer to the FAQ at https://docs.ray.io/en/master/cluster/kubernetes/troubleshooting/troubleshooting.html#kuberay-troubleshootin-guides"
+              f
+              sleep 5
+            don
+            export RAY_HEAD_ADDRESS="${HEAD_SVC}.${POD_NAMESPACE}.svc.cluster.local:6379"
+            echo "Attempting to connect to Ray cluster at $RAY_HEAD_ADDRESS ..."
+            ray start --address="${RAY_HEAD_ADDRESS}" --block
+        env:
+          - name: RAY_USE_TLS
+            value: '1'
+          - name: RAY_TLS_SERVER_CERT
+            value: '/etc/ray/tls/tls.crt'
+          - name: RAY_TLS_SERVER_KEY
+            value: '/etc/ray/tls/tls.key'
+          - name: RAY_TLS_CA_CERT
+            value: '/etc/ca/tls/ca.crt'
+          - name: POD_NAME
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.name
+          - name: POD_IP
+            valueFrom:
+              fieldRef:
+                fieldPath: status.podIP
+          - name: POD_NAMESPACE
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.namespace
+        resources:
+          limits:
+            cpu: '16'
+            memory: 48Gi
+          requests:
+            cpu: '8'
+            memory: 24Gi
+        volumeMounts:
+          - name: shm
+            mountPath: /dev/shm
+          - mountPath: /etc/ca/tls
+            name: ray-ca-cert
+            readOnly: true
+          - mountPath: /etc/ray/tls
+            name: ray-tls
+          - mountPath: /etc/gen/tls
+            name: gen-tls-script
+        livenessProbe:
+          failureThreshold: 2
+          periodSeconds: 5
+          successThreshold: 1
+          timeoutSeconds: 15
+          exec:
+            command:
+              - bash
+              - -c
+              - |
+                # Check if the registered nodes count matches PIPELINE_PARALLEL_SIZE
+                registered_node_count=$(ray status --address ${HEAD_SVC}.${POD_NAMESPACE}.svc.cluster.local:6379 | grep -c node_)
+                if [[ ${registered_node_count} -ne "${PIPELINE_PARALLEL_SIZE}" ]]; then
+                  echo "Unhealthy - Registered nodes count (${registered_node_count}) does not match PIPELINE_PARALLEL_SIZE (${PIPELINE_PARALLEL_SIZE})."
+                  exit 1
+                fi
+        startupProbe:
+          failureThreshold: 40
+          periodSeconds: 30
+          successThreshold: 1
+          timeoutSeconds: 30
+          initialDelaySeconds: 20
```

Comment:
Are these going to be enough time for larger models that take longer to be ready to serve?

Reply:
This wait for minimum 20mins and maximum 30mins. I heard that llama 405b took 12mins to deploy.